mautic/mautic
# Add support for relative dates to campaign action - update contact

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/mautic-documentation/pull/132
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | fixes https://github.com/mautic/mautic/issues/6080
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added support for relative dates in contact update campaign condition You are able to set -5 days or +1 day to any date field to update.

For devs:

This PR added new option `ignore_date_type `(https://github.com/mautic/mautic/pull/6126/files#diff-18ab1b8c93013f1e1d5134f39010ac2aR132) to use form date field without transform date on save.


#### Steps to test this PR:
1. Create custom date field 
2. Create campaign action Update contact
3. Set today, +1 day or date 
4. Create condition with custom date field and set conditon  with relative date (less then -1 day)
5. Check if contact was update and condition works properly during the mautic campaign trigger action 
`php app/console m:c:t`

Update contact action
![image](https://user-images.githubusercontent.com/462477/42207195-9d14731e-7ea9-11e8-8a77-3aa670b6286d.png)

Contact conditon

![image](https://user-images.githubusercontent.com/462477/42207223-b49c71bc-7ea9-11e8-8c47-acc68a9c0a7d.png)
